### PR TITLE
GODRIVER-2406 add FLE 2 doc examples

### PR DESCRIPTION
**Pull request by kevinAlbs**
Test the following:
- Create an encryptedFieldsMap with an indexed and unindexed field.
- Create an FLE 2 collection.
- Auto encrypt an insert and find.

### CODE REVIEW

File: mongo/integration/client_side_encryption_test.go [hunk 1]
```diff
@@ -460,3 +460,128 @@ func TestFLE2CreateCollection(t *testing.T) {
...
+			assert.Nil(mt, err, "error in Connect: %v", err)
+			// Create the FLE 2 collection docsExample.encrypted.
+			db := encryptedClient.Database("docsExamples")
+			// Because docsExample.encrypted is in encryptedFieldsMap, it is created with FLE 2 support.
+			err = db.CreateCollection(context.Background(), "encrypted")
+			assert.Nil(mt, err, "error in CreateCollection")
+			encryptedColl = db.Collection("encrypted")
+		}
+
+		// Auto encrypt an insert and find.
```
Comment (terakilobyte) [inline mongo/integration/client_side_encryption_test.go:551]:
```suggestion
		// Automatically encrypted insert and find operations.
```

File: mongo/integration/client_side_encryption_test.go [hunk 2]
```diff
@@ -460,3 +460,128 @@ func TestFLE2CreateCollection(t *testing.T) {
...
+		{
+			// Encrypt an insert.
```
Comment (terakilobyte) [inline mongo/integration/client_side_encryption_test.go:553]:
```suggestion
			// Insert a document with the encrypted client
```

File: mongo/integration/client_side_encryption_test.go [hunk 3]
```diff
@@ -460,3 +460,128 @@ func TestFLE2CreateCollection(t *testing.T) {
...
+			_, err := encryptedColl.InsertOne(context.Background(), bson.M{
+				"_id":                1,
+				"encryptedIndexed":   "indexedValue",
+				"encryptedUnindexed": "unindexedValue",
+			})
+			assert.Nil(mt, err, "error in InsertOne")
+
+			// Encrypt a find.
```
Comment (terakilobyte) [inline mongo/integration/client_side_encryption_test.go:561]:
```suggestion
			// Find a document with the encrypted client.
```

File: mongo/integration/client_side_encryption_test.go [hunk 4]
```diff
@@ -460,3 +460,128 @@ func TestFLE2CreateCollection(t *testing.T) {
...
+			})
+			assert.Nil(mt, res.Err(), "error in FindOne: %v", res.Err())
+			var resBSON bson.M
+			err = res.Decode(&resBSON)
+			assert.Nil(mt, err, "error in Decode: %v", err)
+			assert.Equal(mt, resBSON["encryptedIndexed"], "indexedValue", "expected 'indexedValue', got %q", resBSON["encryptedIndexed"])
+			assert.Equal(mt, resBSON["encryptedUnindexed"], "unindexedValue", "expected 'unindexedValue', got %q", resBSON["encryptedUnindexed"])
+		}
+
+		// Find documents without decryption.
```
Comment (terakilobyte) [inline mongo/integration/client_side_encryption_test.go:573]:
```suggestion
		// Find a document with an unencrypted client.
```

**Review by terakilobyte — APPROVED** [review]
Small tweaks I would make left as suggestions, let me know if you're looking for other feedback.

**Comment by kevinAlbs** [comment]
Closing for https://github.com/mongodb/mongo-go-driver/pull/969

**Pull request closed without merging**